loss_functions.py

```diff
@@ -155,48 +130,53 @@ def hypo_weight_loss(model_output):
     return weight_sum * (1 / total_weights)
 
 
-def hyper_loss(model_output, gt_sdf, gt_normals, ks, ki, kn, kg):
+def hyper_loss(model_output, gt_sdf, ks, ki, kg, gt_normals=None, kn=None, eps=0.0):
+    losses = {}
+    
     gt_sdf = gt_sdf.unsqueeze(0).float()
-    gt_normals = gt_normals.unsqueeze(0).float()
+
+    if gt_normals is not None:
+        gt_normals = gt_normals.unsqueeze(0).float()
 
     coords = model_output["model_in"].float()
     pred_sdf = model_output["model_out"].float()
 
     sdf_constraint = torch.where(
-        gt_sdf == 0, torch.abs(pred_sdf), torch.zeros_like(pred_sdf)
+        torch.abs(gt_sdf) <= eps, torch.abs(pred_sdf), torch.zeros_like(pred_sdf)
```

Comment:
Please revert this back to the original. We don't want to |s| < eps to be all 0.